File: WCF/Windows SDK WCF Samples/TechnologySamples Beta 1/Basic/Service/ServiceHost/VB/service/service.vb
' Copyright (c) Microsoft Corporation.  All Rights Reserved.

Imports System
Imports System.ServiceModel.Activation
Imports System.ServiceModel

Namespace Microsoft.ServiceModel.Samples

    ' Define a service contract.
    <ServiceContract([Namespace]:="http://Microsoft.ServiceModel.Samples")> _
    Public Interface ICalculator

        <OperationContract()> _
        Function Add(ByVal n1 As Double, ByVal n2 As Double) As Double
        <OperationContract()> _
        Function Subtract(ByVal n1 As Double, ByVal n2 As Double) As Double
        <OperationContract()> _
        Function Multiply(ByVal n1 As Double, ByVal n2 As Double) As Double
        <OperationContract()> _
        Function Divide(ByVal n1 As Double, ByVal n2 As Double) As Double

    End Interface

    ' Service class which implements the service contract.
    Public Class CalculatorService
        Implements ICalculator

        Public Function Add(ByVal n1 As Double, ByVal n2 As Double) As Double Implements ICalculator.Add

            Return n1 + n2

        End Function

        Public Function Subtract(ByVal n1 As Double, ByVal n2 As Double) As Double Implements ICalculator.Subtract

            Return n1 - n2

        End Function

        Public Function Multiply(ByVal n1 As Double, ByVal n2 As Double) As Double Implements ICalculator.Multiply

            Return n1 * n2

        End Function

        Public Function Divide(ByVal n1 As Double, ByVal n2 As Double) As Double Implements ICalculator.Divide

            Return n1 / n2

        End Function

    End Class

    ' Implement a ServiceHostFactoryBase to control initialization of a Service
    ' Service.svc file is modified to activate this ServiceHost
    Public Class CalculatorServiceHostFactory
        Inherits ServiceHostFactoryBase

        Public Overloads Overrides Function CreateServiceHost(ByVal constructorString As String, ByVal baseAddresses As Uri()) As ServiceHostBase

            Dim serviceHost As New ServiceHost(GetType(CalculatorService), baseAddresses)

            ' Remove all configured endpoints
            serviceHost.Description.Endpoints.Clear()

            ' Add an endpoint that supports BasicHttpBinding
            serviceHost.AddServiceEndpoint(GetType(ICalculator), New BasicHttpBinding(), "")
            ' Add a mex endpoint
            serviceHost.AddServiceEndpoint("IMetadataExchange", System.ServiceModel.Description.MetadataExchangeBindings.CreateMexHttpBinding(), "/mex")
            Return serviceHost

        End Function

    End Class

End Namespace

File: WCF/Windows SDK WCF Samples/TechnologySamples Beta 1/Basic/Service/ServiceHost/CS/service/service.cs

//  Copyright (c) Microsoft Corporation.  All Rights Reserved.

using System;
using System.ServiceModel.Activation;
using System.ServiceModel;

namespace Microsoft.ServiceModel.Samples
{
    // Define a service contract.
    [ServiceContract(Namespace="http://Microsoft.ServiceModel.Samples")]
    public interface ICalculator
    {
        [OperationContract]
        double Add(double n1, double n2);
        [OperationContract]
        double Subtract(double n1, double n2);
        [OperationContract]
        double Multiply(double n1, double n2);
        [OperationContract]
        double Divide(double n1, double n2);
    }

    // Service class which implements the service contract.
    public class CalculatorService : ICalculator
    {
        public double Add(double n1, double n2)
        {
            return n1 + n2;
        }

        public double Subtract(double n1, double n2)
        {
            return n1 - n2;
        }

        public double Multiply(double n1, double n2)
        {
            return n1 * n2;
        }

        public double Divide(double n1, double n2)
        {
            return n1 / n2;
        }
    }

    // Implement a ServiceHostFactoryBase to control initialization of a Service
    // Service.svc file is modified to activate this ServiceHost
    public class CalculatorServiceHostFactory : ServiceHostFactoryBase
    {
        public override ServiceHostBase CreateServiceHost(string constructorString, Uri[] baseAddresses)
        {
            ServiceHost serviceHost = new ServiceHost(typeof(CalculatorService), baseAddresses);
          
            // Remove all configured endpoints
            serviceHost.Description.Endpoints.Clear();

            // Add an endpoint that supports BasicHttpBinding
            serviceHost.AddServiceEndpoint(typeof(ICalculator), new BasicHttpBinding(), "");
            // Add a mex endpoint
            serviceHost.AddServiceEndpoint("IMetadataExchange", System.ServiceModel.Description.MetadataExchangeBindings.CreateMexHttpBinding(), "/mex");
            return serviceHost;
        }

    }    
   
}
